### PR TITLE
Migration script V1 -> V2

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This is a draft migration script to mover artifacts from V1 to V2. It's a little untested because I'd like to make sure it works in CI first. It's obviously something which could go very wrong if it's not set up right.

This PR doesn't include any form of documentation on how to use it yet - I'm going to propose that I follow up with that after actually successfully managing a migration.

### CODE REVIEW

**Comment by NiallRees** [comment]
Let me know when out of draft @alanmcruickshank

File: macros/migrate_v1_to_v2.sql [hunk 1]
```diff
@@ -0,0 +1,117 @@
+{% macro migrate_artifacts_v1_to_v2() %}
+
+{% set src_dbt_artifacts = source('dbt_artifacts', 'artifacts') %}
+{% set src_results = source('dbt_artifacts', 'dbt_run_results') %}
+{% set src_results_nodes = source('dbt_artifacts', 'dbt_run_results_nodes') %}
+{% set src_manifest_nodes = source('dbt_artifacts', 'dbt_run_manifest_nodes') %}
```
Comment (NiallRees) [inline macros/migrate_v1_to_v2.sql]:
```suggestion
{% set src_manifest_nodes = source('dbt_artifacts', 'dbt_manifest_nodes') %}

```

Just noticed this, may need updating elsewhere. Nitpick but just avoids confusion, the artifacts are named 'run_results' and 'manifest'

Comment (alanmcruickshank) [reply macros/migrate_v1_to_v2.sql]: Good call - it does impact a few things to change - but much easier to change at this stage than later down the line.

**Comment by alanmcruickshank** [comment]
Tested live and working 👍 .

Our project now has a 13GB `dbt_manifest_nodes` table with 77M rows 🚀

**Review by NiallRees — APPROVED** [review]

**Review by Jims55jr — CHANGES_REQUESTED** [review]
HELLO

**Pull request merged**